docs/source/format/index.rst

```diff
@@ -22,17 +22,38 @@ Specifications
 
 .. toctree::
    :maxdepth: 2
+   :caption: Arrow Columnar Format
+   :hidden:
 
-   Versioning
+   FormatIntro
    Columnar
+   Versioning
+   Changing
    CanonicalExtensions
    Other
+
+.. toctree::
+   :maxdepth: 2
+   :caption: C definitions
+   :hidden:
+
    CDataInterface
    CStreamInterface
    CDeviceDataInterface
```

Comment:
@jorisvandenbossche have kept C Stream Interface in a separate file as the structure is nicer IMHO.